### PR TITLE
Tools/MMapsGenerator: percentage progress

### PR DESCRIPTION
**Pull request by IntelligentQuantum**
## Changes Proposed:
- 
-  

## Issues Addressed:

- Closes 

## SOURCE:

https://github.com/TrinityCore/TrinityCore/commit/b9f276f0a00e2f546bc40875a2d809b7c633c495
https://github.com/TrinityCore/TrinityCore/commit/29937aaa750ae5485ab20de231399f83b999ca49
## Tests Performed:

- Need Test
- Part of multi threading mmaps
- re-extract required? NO

## How to Test the Changes:

1. checked that percentage only grows and doesnt shift back and forth between two values
2. checked that percentage does reach 99% on both a single map and all maps run
3. Built mmaps with old and new extractor and took md5 hashes of the mmap file contents to verify that the output was not changed. Also compared the folder sizes which matched
4.Verified moving tile discovering to single threaded part is not spending noticeable amount of time

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by temperrr** [comment]
How to test

**Comment by acidmanifesto** [comment]
> How to test

the only way to test it is to actualy view the map extraction process. while is extracts we will get a % of what is done and on what map while it extracts

**Comment by IntelligentQuantum** [comment]
@pklloveyou Can you test this pr please, i need it for continue my work

**Comment by Lynethris** [comment]
PR Tested (it's was  long), but it's work : 

- [X] Percentage only growth
- [X] Percentage does reach 99% on both a single map and all maps run
- [X] MD5 Checksum is the same than before

File: src/tools/mmaps_generator/MapBuilder.cpp [hunk 1]
```diff
@@ -904,6 +888,9 @@ namespace MMAP
     /**************************************************************************/
     bool MapBuilder::shouldSkipMap(uint32 mapID)
     {
+        if (m_mapid >= 0)
+            return static_cast<uint32>(m_mapid) != mapID;
```
Comment (Winfidonarleyan) [inline src/tools/mmaps_generator/MapBuilder.cpp:892]: brackets

**Review by Winfidonarleyan — APPROVED** [review]

**Pull request merged**